src/Microsoft.AspNet.Diagnostics/ErrorPageExtensions.cs

```diff
@@ -31,7 +31,7 @@ public static IApplicationBuilder UseErrorPage([NotNull] this IApplicationBuilde
         /// <returns></returns>
         public static IApplicationBuilder UseErrorPage([NotNull] this IApplicationBuilder builder, ErrorPageOptions options)
         {
-            return builder.Use(next => new ErrorPageMiddleware(next, options).Invoke);
+            return builder.UseMiddleware<ErrorPageMiddleware>(options);
```

Comment:
I have tested this change where it populates the logger factory parameter, so i think this change is fine